src/amo/pages/AddonReviewList/index.js

```diff
@@ -333,29 +333,36 @@ export class AddonReviewListBase extends React.Component<InternalProps> {
           {reviewId && (
             <FeaturedAddonReview addon={addon} reviewId={reviewId} />
           )}
-          <CardList
-            className="AddonReviewList-reviews-listing"
-            footer={paginator}
-            header={reviewCountHTML}
-          >
-            <ul>
-              {allReviews.map((review, index) => {
-                // TODO: Remove this and use the API to filter out the featured review once
-                // https://github.com/mozilla/addons-server/issues/9424 is fixed.
-                if (
-                  !reviewId ||
-                  (review && review.id.toString() !== reviewId)
-                ) {
-                  return (
-                    <li key={String(index)}>
-                      <AddonReviewCard addon={addon} review={review} />
-                    </li>
-                  );
-                }
-                return null;
-              })}
-            </ul>
-          </CardList>
+          {// TODO: remove this condition after
+          //  https://github.com/mozilla/addons-server/issues/9424 is fixed
+          //  and integrated.
+          (!reviewId ||
+            allReviews.length > 1 ||
+            allReviews[0].id.toString() !== reviewId) && (
+            <CardList
+              className="AddonReviewList-reviews-listing"
+              footer={paginator}
+              header={reviewCountHTML}
+            >
+              <ul>
+                {allReviews.map((review, index) => {
```

Comment:
what about using filter and map (or even reduce):

may be better/cleaner to just get params instead of this includes way but this is just an example:

```javascript
const reviewListing = allReviews
      .filter((item) => item && !this.props.location.pathname.includes(item.id))
      .map((review, index) => {
        // TODO: Remove this and use the API to filter out the featured review once
        // https://github.com/mozilla/addons-server/issues/9424 is fixed.

        return (
          <li key={String(index)}>
            <AddonReviewCard addon={addon} review={review} />
          </li>
        );
      });

....

<ul>{reviewsListing}</ul>
```

Reply:
This is nice, but the issue is that we want to hide the entire card.

Reply:
right this is just for the inner mapping code....if that makes sense..sorry if that wasn't clear..

Reply:
Ah, sorry. I think you inspired me to a better solution anyway. :-)